code/controllers/subsystem/movement/movement.dm

```diff
@@ -2,17 +2,28 @@ SUBSYSTEM_DEF(movement)
 	name = "Movement Loops"
 	flags = SS_NO_INIT|SS_BACKGROUND|SS_TICKER
 	wait = 1 //Fire each tick
-	///The list of datums we're processing
-	var/list/processing = list()
-	///Used to make pausing possible
-	var/list/currentrun = list()
+	/*
+		A breif aside about the bucketing system here
+
+		The goal is to allow for higher loads of semi long delays while reducing cpu usage
+		Bucket insertion and management are much less complex then what you might see in SStimer
+		This is intentional, as we loop our delays much more often then that ss is designed for
+		We also have much shorter term timers, so we need to worry about redundant buckets much less
+	*/
+	///Assoc list of "target time" -> list(things to process). Used for quick lookup
+	var/list/buckets = list()
+	///Sorted list of list(target time, bucket to process)
+	var/list/sorted_buckets = list()
 	///The time we started our last fire at
 	var/canonical_time = 0
 	///The visual delay of the subsystem
 	var/visual_delay = 1
 
 /datum/controller/subsystem/movement/stat_entry(msg)
-	msg = "P:[length(processing)]"
+	var/total_len = 0
+	for(var/list/bucket in sorted_buckets)
```

Comment:
```suggestion
	for(var/list/bucket as anything in sorted_buckets)
```
Casting it as a list will still trigger a type check. I assume there's only lists here?

Reply:
It uh, it will? good god

Reply:
```dm
var/list/l = list( list(1, 2), 2, list(3, 4))
world.log << "Type casting:"
for(var/list/t in l)
	world.log << islist(t)
world.log << "No type casting:"
for(var/list/t as anything in l)
	world.log << islist(t)
```
Result:
```
Type casting:
1
1
No type casting:
1
0
1
```

Reply:
:L